tests/test_cli_replace_modal_subgraph.py

```diff
@@ -0,0 +1,195 @@
+    n.write_to_matsim(output_dir)
+    return {"network": n, "link_id": "link_n1_n2", "path": output_dir / "network.xml"}
+
+
+@pytest.fixture()
+def sub_network(tmpdir):
+    output_dir = Path(tmpdir) / "sub_network"
+    n = Network("epsg:27700")
+    n.add_nodes({"n1": {"x": 1, "y": 1}, "n2": {"x": 2, "y": 2}})
+    n.add_link(
+        "link_n1_n2",
+        "n1",
+        "n2",
+        attribs={
+            "modes": {"car", "bike", "bus"},
+            "permlanes": 1,
+            "freespeed": 1,
+            "capacity": 1,
+            "attributes": {"unique_data": "yes"},
+        },
+    )
+    n.write_to_matsim(output_dir)
+    return {
+        "network": n,
+        "bike_link_id": "link_n1_n2",
+        "expected_bike_link_id": "bike---link_n1_n2",
+        "expected_link_id_mapping": {"link_n1_n2": "bike---link_n1_n2"},
+        "path": output_dir / "network.xml",
+    }
+
+
+def test_retains_original_link_strips_bike_mode(tmpdir, network, sub_network):
+    link_id = sub_network["bike_link_id"]
+    assert "bike" in network["network"].link(link_id)["modes"]
+
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike",
+        increase_capacity=True,
+    )
+
+    assert output_network.has_link(link_id), f"Link {link_id} is missing from the output network"
+    assert (
+        "bike" not in output_network.link(link_id)["modes"]
+    ), f"Mode `bike` was not removed from link: {link_id}"
+
+
+def test_adds_new_links_from_sub_network_just_for_bike(tmpdir, network, sub_network):
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike",
+        increase_capacity=True,
+    )
+
+    expected_new_link_id = sub_network["expected_bike_link_id"]
+    assert output_network.has_link(
+        expected_new_link_id
+    ), f"Link {expected_new_link_id} is missing from the output network"
+    assert output_network.link(expected_new_link_id)["modes"] == {
+        "bike"
+    }, f"Link {expected_new_link_id} did not have the correct modes set"
+
+
+def test_retains_link_data_from_subgraph_network(tmpdir, network, sub_network):
+    original_link_data = sub_network["network"].link(sub_network["bike_link_id"])
+
+    output_network = cli._replace_modal_subgraph(
+        path_to_network=network["path"],
+        projection="epsg:27700",
+        output_dir=tmpdir,
+        path_to_subgraph=sub_network["path"],
+        modes="bike",
+        increase_capacity=True,
+    )
+
+    output_data = output_network.link(sub_network["expected_bike_link_id"])
+    keys_to_ignore = {"from", "s2_from", "to", "s2_to", "id", "modes", "capacity"}
+    assert {k: v for k, v in output_data.items() if k not in keys_to_ignore} == {
```

Comment:
Maybe this assertion can be extracted into a helper method like `assert_link_data_equal(link_1, link_2)` that would keep the intent of the assertion clear but remove some of the noisy detail of the comparison mechanics and make the test a little easier to understand.